lib/file.c

```diff
@@ -419,10 +421,12 @@ static CURLcode file_do(struct Curl_easy *data, bool *done)
         return result;
     }
 
-    result = Curl_client_write(data, CLIENTWRITE_HEADER,
-                               (char *)"Accept-ranges: bytes\r\n", 0);
-    if(result)
-      return result;
+    if(!S_ISDIR(statbuf.st_mode)) {
```

Comment:
I wouldn't execute that function again. Checking the now set variable expected_size is more performant

Reply:
POSIX defines this as a macro, so the performance impact of doing a bit test should be negligible.

In any case, now we leave it initialized at -1 this can move up into the previous if statement anyway, which means we don't need to check anything at all!